src/include/common/arrow/arrow_row_batch.h

```diff
@@ -44,46 +44,46 @@ struct ArrowVector {
 
     //! Append a data chunk to the underlying arrow array
     ArrowArray append(main::QueryResult& queryResult, std::int64_t chunkSize);
 
 private:
-    static std::unique_ptr<ArrowVector> createVector(const main::DataTypeInfo& typeInfo,
+    static std::unique_ptr<ArrowVector> createVector(const LogicalType& type,
         std::int64_t capacity);
-    static void appendValue(ArrowVector* vector, const main::DataTypeInfo& typeInfo, Value* value);
+    static void appendValue(ArrowVector* vector, const LogicalType& type, Value* value);
 
-    static ArrowArray* convertVectorToArray(ArrowVector& vector,
-        const main::DataTypeInfo& typeInfo);
-    static ArrowArray* convertStructVectorToArray(ArrowVector& vector,
-        const main::DataTypeInfo& typeInfo);
+    static ArrowArray* convertVectorToArray(ArrowVector& vector, const LogicalType& type);
+    static ArrowArray* convertStructVectorToArray(ArrowVector& vector, const LogicalType& type);
+    static ArrowArray* convertInternalIDVectorToArray(ArrowVector& vector, const LogicalType& type);
     static inline void initializeNullBits(ArrowBuffer& validity, std::int64_t capacity) {
         auto numBytesForValidity = getNumBytesForBits(capacity);
         validity.resize(numBytesForValidity, 0xFF);
     }
-    static void initializeStructVector(ArrowVector* vector, const main::DataTypeInfo& typeInfo,
+    static void initializeStructVector(ArrowVector* vector, const LogicalType& type,
+        std::int64_t capacity);
+    static void initializeInternalIDVector(ArrowVector* vector, const LogicalType& type,
         std::int64_t capacity);
-    static void copyNonNullValue(ArrowVector* vector, const main::DataTypeInfo& typeInfo,
-        Value* value, std::int64_t pos);
+    static void copyNonNullValue(ArrowVector* vector, const LogicalType& type, Value* value,
+        std::int64_t pos);
     static void copyNullValue(ArrowVector* vector, Value* value, std::int64_t pos);
 
     template<LogicalTypeID DT>
-    static void templateInitializeVector(ArrowVector* vector, const main::DataTypeInfo& typeInfo,
+    static void templateInitializeVector(ArrowVector* vector, const LogicalType& type,
         std::int64_t capacity);
     template<LogicalTypeID DT>
-    static void templateCopyNonNullValue(ArrowVector* vector, const main::DataTypeInfo& typeInfo,
-        Value* value, std::int64_t pos);
+    static void templateCopyNonNullValue(ArrowVector* vector, const LogicalType& type, Value* value,
+        std::int64_t pos);
     template<LogicalTypeID DT>
     static void templateCopyNullValue(ArrowVector* vector, std::int64_t pos);
     template<LogicalTypeID DT>
-    static ArrowArray* templateCreateArray(ArrowVector& vector, const main::DataTypeInfo& typeInfo);
+    static ArrowArray* templateCreateArray(ArrowVector& vector, const LogicalType& type);
 
     ArrowArray toArray();
 
 private:
-    std::vector<std::unique_ptr<main::DataTypeInfo>> typesInfo;
+    common::logical_types_t types;
```

Comment:
use `std::vector<common::LogicalType>`. I'm planning to remove `common::logical_types_t`